src/app/services/reducers/query-runner-reducers.ts

```diff
@@ -1,12 +1,23 @@
 import { IAction } from '../../../types/action';
-import { QUERY_GRAPH_SUCCESS } from '../redux-constants';
+import { QUERY_GRAPH_ERROR, QUERY_GRAPH_RUNNING, QUERY_GRAPH_SUCCESS } from '../redux-constants';
 
 export function graphResponse(state = {}, action: IAction): object {
   switch (action.type) {
     case QUERY_GRAPH_SUCCESS:
       if (typeof action.response !== 'string') {
         return action.response;
       }
+    case QUERY_GRAPH_ERROR:
+      if (typeof action.response !== 'string') {
```

Comment:
whay are we checking if payload  is a not of type string

Reply:
Payloads can either be string or objects. 

The other alternative is for the return type to be `object|string` instead of  `object`

Reply:
@jobala but the action creators have response type as `object`  `queryResponseError(response: object)`, `queryRunningStatus(response: object)` so we can be sure  if the case is `QUERY_GRAPH_ERROR` then we have an object, but you are saying that `response.body` `queryResponseError(response.body)` might be a string or object.

Reply:
Action creators are of the type `IAction`. If you look at `src/types/action.ts`, you will see that the `response` property is either an object or string.